tmate/install.sh

```diff
@@ -109,22 +124,40 @@ fi
 sed -i.bak "s|localhost:8080|$1|g" $SCRIPTPATH/controller.py
 sed -i.bak "s|localhost:8000|$1|g" $SCRIPTPATH/.tmate.conf
 
+# FIXME: seems like this todo is done
 # TODO: make the fun timer stuff in provision.sh work without needing to run it...
 
-# sudo install -m 755 -d /opt/hc-client
 # sudo install -m 755 $SCRIPTPATH/{connected.py,deprovision.sh,device.py,provision.sh} /opt/hc-client
 
 
-sudo install -m 755 $SCRIPTPATH/controller.py $APP_PATH
-sudo install -m 644 $SCRIPTPATH/.tmate.conf /home/$UNAME/.tmate.conf
-sudo install -m 644 $SCRIPTPATH/{session.target,session@.service,controller.service} /etc/systemd/system
+sudo -u $UNAME install -m 644 $SCRIPTPATH/.tmate.conf /home/$UNAME/.tmate.conf
+sudo install -m 755 $SCRIPTPATH/controller.py $APP_PATH/
+
+if [[ "${INIT}" == "systemd" ]]; then
+    sudo install -m 644 $SCRIPTPATH/{session.target,session@.service,controller.service} /etc/systemd/system/ || die "couldn't install systemd stuff"
+fi
+
+if [[ "${INIT}" == "upstart" ]]; then
+    sudo install -m 644 $SCRIPTPATH/{session.conf,controller.conf} /etc/init/ || die "couldn't install upstart stuff"
+fi
 
 #Make .bashrc immutable
-echo -e "\nunset AUTH\n" >> /home/$UNAME/.bashrc
-chattr +i /home/$UNAME/.bashrc
+if ! grep -q "unset AUTH" /home/$UNAME/.bashrc; then
+echo -e "\nunset AUTH\n" | sudo -u $UNAME tee -a /home/$UNAME/.bashrc
+sudo chattr +i /home/$UNAME/.bashrc
+fi
 
-sudo $SCRIPTPATH/create_config.py $2 6
+sudo $SCRIPTPATH/create_config.py $2 "${NUM_SESSIONS}"
+
+if [[ "${INIT}" == "systemd" ]]; then
```

Comment:
Why not coalesce this with the previous checks against systemd?

Reply:
was just trying to keep context preserved as much as possible. happy to merge it